File: fizzbuzz/java/fizzbuzz.java
public class fizzbuzz {
  public static void main(String[] args) {
    int num = 1;
    String str = "";
    while(num < 101)
    {
      str = "";
      if(num % 3 == 0)
      {//modulo operator returns 0 if num is divisible by 3
        str += "Fizz";
      }
      if(num % 5 == 0){str += "Buzz";}
      if(str.length() > 0)
      {//length is the only built in function besides println and some basic logic and increments
        System.out.println(str);
      }
      else{System.out.println(num);}
      num++;
    }
  }
}

File: fizzbuzz/c++/fizzbuzz.cpp
#include <stdio.h>

int main()
{
    int current_number = 1;
    bool divisible_by_three, divisible_by_five, divisible_by_both;
    while(current_number < 101)
    {
        divisible_by_three = (current_number % 3) == 0;
        divisible_by_five = (current_number % 5) == 0;
        divisible_by_both = divisible_by_three && divisible_by_five;
        if(divisible_by_both)
        {
            printf("FizzBuzz\n");
            current_number++;
            continue;
        }
        if(divisible_by_three)
        {
            printf("Fizz\n");
            current_number++;
            continue;
        }
        if(divisible_by_five)
        {
            printf("Buzz\n");
            current_number++;
            continue;
        }
        printf("%i\n", current_number++);
    }
    return 0;
}
